src/main/java/org/openstreetmap/atlas/checks/distributed/ShardedIntegrityChecksSparkJob.java

```diff
@@ -0,0 +1,434 @@
+                                    this.countryChecks.get(countryShard.getKey())))
+                            .collect(Collectors.toList());
+
+                    // Set spark UI job title
+                    this.getContext().setJobGroup("0",
+                            String.format("Running checks on %s",
+                                    tasksForCountry.get(0).getCountry()));
+                    // Run each task in spark, producing UniqueCheckFlagContainers
+                    final JavaPairRDD<String, UniqueCheckFlagContainer> rdd = this.getContext()
+                            .parallelize(tasksForCountry, tasksForCountry.size())
+                            .mapToPair(produceFlags(input, output, this.configurationMap(),
+                                    fileHelper, shardingBroadcast, distanceToLoadShards,
+                                    (Boolean) commandMap.get(MULTI_ATLAS)));
+                    // Save the RDD
+                    rdd.persist(storageLevel);
+                    // Use a fast spark action to overcome spark lazy elocution. This is necessary
+                    // to get the UI title to appear in the right spot.
+                    rdd.count();
+                    // Add the RDDs to the list
+                    countryRdds.add(rdd);
+                });
+            }
+        }
+
+        // Set up for unioning
+        final JavaPairRDD<String, UniqueCheckFlagContainer> firstCountryRdd = countryRdds.get(0);
+        countryRdds.remove(0);
+
+        // Add UI title
+        this.getContext().setJobGroup("0", "Conflate flags and generate outputs");
+        // union the RDDs, combine the UniqueCheckFlagContainers by country, and proccess the flags
+        // through the output event service
+        this.getContext().union(firstCountryRdd, countryRdds)
+                .reduceByKey(UniqueCheckFlagContainer::combine)
+                .foreach(processFlags(output, fileHelper, outputFormats));
+
+        logger.info("Sharded checks completed in {}", start.elapsedSince());
+    }
+
+    @Override
+    protected SwitchList switches()
+    {
+        return super.switches().with(EXPANSION_DISTANCE, MULTI_ATLAS, SPARK_STORAGE_DISK_ONLY,
+                SHARDING);
+    }
+
+    /**
+     * Get the fetcher to use for Atlas files. The fetcher uses a hadoop cache to reduce remote
+     * reads.
+     *
+     * @param input
+     *            {@link String} input folder path
+     * @param country
+     *            {@link String} country code
+     * @param configuration
+     *            {@link org.openstreetmap.atlas.generator.tools.spark.SparkJob} configuration map
+     * @return {@link Function} that fetches atlases/
+     */
+    private Function<Shard, Optional<Atlas>> atlasFetcher(final String input, final String country,
+            final Map<String, String> configuration)
+    {
+        final HadoopAtlasFileCache cache = new HadoopAtlasFileCache(input, configuration);
```

Comment:
I believe making a new one of these WILL work for every atlas fetcher, but if you can make it a transient static lazy initialized I believe that will mean each JVM working in the spark cluster will use its own, which may allow for optimizations.

Reply:
The caches are using a global namespace, so I think this should already effectively be one cache store per machine in the cluster.